WalletWasabi/Extensions/TaskExtensions.cs

```diff
@@ -78,12 +78,23 @@ public static async Task WithAwaitCancellationAsync(this Task me, TimeSpan timeo
 			=> await me.WithAwaitCancellationAsync(ThreadingHelpers.Cancelled, (int)timeout.TotalMilliseconds).ConfigureAwait(false);
 
 		public static async Task<TResult> RunAsScheduledAsync<TResult>(this Task<TResult> task, DateTimeOffset when, CancellationToken cancellationToken)
+		{
+			await WaitUntilAsync(when, cancellationToken).ConfigureAwait(false);
+			return await task.ConfigureAwait(false);
+		}
+
+		public static async Task RunAsScheduledAsync(this Task task, DateTimeOffset when, CancellationToken cancellationToken)
```

Comment:
There are a few places where we have this duplicated methods for `Task<TResult>` and `Task` because we don't have an `Unit` type. This is not something unknown by the UI team but it is somehow a new concept in the backend code. Would you agree if I added it and use it to simplify the code (not only here)?

Reply:
ACK